client/my-sites/upgrades/cart/cart-body.jsx

```diff
@@ -1,50 +1,56 @@
 /**
  * External dependencies
  */
-var React = require( 'react' );
+import React, { Component, PropTypes } from 'react';
+import { localize } from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
-var CartItems = require( './cart-items' ),
-	CartCoupon = require( './cart-coupon' ),
-	CartTotal = require( './cart-total' );
-
-var CartBody = React.createClass( {
-	propTypes: {
-		collapse: React.PropTypes.bool
-	},
-	getDefaultProps: function() {
-		return {
-			collapse: false,
-			showCoupon: false
-		};
-	},
-
-	render: function() {
-		if ( ! this.props.cart.hasLoadedFromServer ) {
-			return <div className="cart-body">{ this.translate( 'Loading…', { context: 'Upgrades: Loading cart' } ) }</div>;
+import CartItems from './cart-items';
+import CartCoupon from './cart-coupon';
+import CartTotal from './cart-total';
+
+class CartBody extends Component {
+	static propTypes = {
+		collapse: PropTypes.bool
+	};
+
+	static defaultProps = {
+		collapse: false,
+		showCoupon: false
+	};
+
+	optionalCoupon() {
+		if ( ! this.props.showCoupon ) {
+			return;
+		}
+
+		return <CartCoupon cart={ this.props.cart } />;
+	}
+
+	render() {
+		const { translate, cart } = this.props;
+
+		if ( ! cart.hasLoadedFromServer ) {
+			return <div className="cart-body">
+				{ translate( 'Loading…', { context: 'Upgrades: Loading cart' } ) }
```

Comment:
oh. I was unaware that we had comments that weren't context. could you provide a quick snippet showing that @deBhal ?